parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -420,9 +421,25 @@ extractKubeBinaries() {
         k8s_tgz_tmp="${k8s_downloads_dir}/${k8s_tgz_tmp_filename}"
         mkdir -p ${k8s_downloads_dir}
 
-        retrycmd_get_tarball 120 5 "${k8s_tgz_tmp}" ${kube_binary_url} || exit $ERR_K8S_DOWNLOAD_TIMEOUT
-        if [[ ! -f ${k8s_tgz_tmp} ]]; then
-            exit "$ERR_K8S_DOWNLOAD_TIMEOUT"
+        registry_regex='^.+\/.+\/.+:.+$'
+        if [[ ${kube_binary_url} =~ $registry_regex ]]; then
+            echo "detect kube_binary_url, ${kube_binary_url}, as registry url, will use oras to pull artifact binary"
+            # download the kube package from registry as oras artifact
+            if [[ $(isARM64) == 1 ]]; then
```

Comment:
You can reference my WIP PR - https://github.com/Azure/AgentBaker/pull/4879/files#diff-59521d60320265e88e46f4702029a64c03d857baee2923ff450f4997f6184fe8

for how I set CPU arch to make it a bit cleaner